client/components/token-field/index.jsx

```diff
@@ -463,16 +474,24 @@ var TokenField = React.createClass( {
 		} );
 	},
 
-	_addNewToken: function( token ) {
-		var newValue;
+	_addNewToken: function( token, { isBatchOperation = false } = {} ) {
+		const tokens = Array.isArray( token ) ? token : [ token ];
+		const newValue = clone( this.props.value );
+		let newValueIndex = this._getIndexOfInput();
+
+		tokens.forEach( token => {
+			token = this.props.saveTransform( token );
 
-		token = this.props.saveTransform( token );
+			if ( ! this._valueContainsToken( token ) ) {
```

Comment:
There is a bug here: we can add the same value multiple times since `this.props.value` is not updated.  I'll fix it in a couple of hours.